standards.md

```diff
@@ -0,0 +1,315 @@
+|------|------------|
+| CMEC_CODE_DIR <br>MDTF_POD_CODE_DIR| The directory where the DEM’s code can be found |
+| CMEC_OBS_DATA<br>MDTF_POD_OBS_DATA| The directory where the relevant observational data can be found. |
+| CMEC_MODEL_DATA<br>MDTF_MODEL_DATA | The directory containing the model input data.  If autocurated metadata has been generated, it can be found in the file autocurator.json in the root directory. |
+| CMEC_WK_DIR<br>MDTF_POD_WK_DIR | The working directory for output from the DEM. |
+| MDTF_CASENAME<br>MDTF_FIRSTYR<br>MDTF_LASTYR | (optional) Variables used by MDTF PODs for labeling plots. |
+| CMEC_DEBUG<br>MDTF_DEBUG | (optional) If set to 1 then the DEM should output more verbose logging information to help with debugging. |
+| CMEC_SAVE_NC<br>MDTF_SAVE_NC | (optional) If set to 0 then the DEM should skip any steps where raw output data is saved in netCDF format. |
+
+<p>&nbsp;</p>
+
+The following restrictions hold on DEM operation:  
+
+DEMs should not refer to, or attempt to access, any files outside of the directories specified by the environment variables above.
+All files in *_CODE_DIR, *_OBS_DATA, or *_MODEL_DATA should be treated as read-only.  
+
+Output from DEMs should only be written to *_WK_DIR. Any intermediate files not part of the output bundle should be deleted before exiting. DEMs should not assume persistence of any files across runs.  
+
+Temporary files, if needed, can be written to the working directory, preferably in the “temp” subfolder. This directory should be removed once execution is completed successfully.  
+
+As a best practice, DEMs should not attempt to open network connections or access resources on remote hosts.  
+
+DEM output can be logged to stdout or stderr. Both outputs may not be displayed to the user directly, but will be logged to a run log.
+
+## Common Output Bundle Format <a name="commonoutput"></a>
+
+Output must be accompanied by a JSON file describing what is present in the output directory. It is recommended that the DEM has defaults in place for circumstances where observation or model metadata is missing.
+
+Base level keys (at least one should be specified):
+| Key | Definition |
+|------|------|
+| index | Short name of the plot/html/metric that should be opened when the user chooses to “open” the output bundle. |
+| provenance | Command line and version information used to execute the DEM. This includes environment variables and the observational datasets used (including dataset versions).  |
+| data | (optional) Dictionary of data files produced with the output. |
+| plots | (optional) Dictionary of plots produced with the output. |
+| html | (optional) Dictionary of html files produced with the output. |
+| metrics | (optional) Dictionary of metric files produced with the output.  |
+
+<p>&nbsp;</p>
+
+Required provenance keys:
+| Key  | Definition |
+|------|------------|
+| environment | Key/value pairs listing all relevant environment variables. |
```

Comment:
What does "relevant" mean in this context?  I am not sure I can answer that.

Reply:
@jkrasting I took a shot at rephrasing this. Let me know if the edit makes sense. I'm not sure if it captures the original intent here or not.

Reply:
@paullric Can you weigh in on this one? The new phrasing I've proposed in "Key/value pairs listing important packages in the run environment and their version."

Reply:
@acordonez - what about "Key/value pairs listing all relevant diagnostic and framework environment variables." ?